source/include/azure_iot_hub_client.h

```diff
@@ -0,0 +1,411 @@
+                                                                   void * prvCallbackContext,
+                                                                   uint32_t ulTimeoutMilliseconds );
+
+/**
+ * @brief Subscribe to device twin.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @param[in] xTwinCallback The #AzureIoTHubClientTwinCallback_t to invoke when device twin messages arrive.
+ * @param[in] prvCallbackContext A pointer to a context to pass to the callback.
+ * @param[in] ulTimeoutMilliseconds Timeout in millisecond for Subscribe operation to complete.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_DeviceTwinSubscribe( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle,
+                                                                 AzureIoTHubClientTwinCallback_t xTwinCallback,
+                                                                 void * prvCallbackContext,
+                                                                 uint32_t ulTimeoutMilliseconds );
+
+/**
+ * @brief Unsubscribe from cloud to device messages.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_CloudMessageUnsubscribe( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle );
+
+/**
+ * @brief Unsubscribe from direct methods.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_DirectMethodUnsubscribe( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle );
+
+/**
+ * @brief Unsubscribe from device twin.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_DeviceTwinUnsubscribe( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle );
+
+/**
+ * @brief Send a response to a received direct method message.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @param[in] pxMessage The pointer to the #AzureIoTHubClientMethodRequest_t to which a response is being sent.
+ * @param[in] ulStatus A code that indicates the result of the method, as defined by the user.
+ * @param[in] pucMethodPayload __[nullable]__ An optional method response payload.
+ * @param[in] ulMethodPayloadLength The length of the method response payload.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_SendMethodResponse( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle,
```

Comment:
good catch agree

Reply:
either that or can we shorten and just use method everywhere and drop direct?

Reply:
Oh yes there's a non-zero chance this will get renamed to SendCommandResponse as part of the "PnP-api-ification(TM)" of the world and especially since FreeRTOS doesn't have legacy concerns.  So I think we get consistent & pretty close with legacy but I don't think we need to spend 40 hours arguing over it.  Not that I'd want that anyway...